### DISCUSSION TITLE
Nice name, bro

### DISCUSSION THREAD

**Discussion by reverofevil** [tamius-han/ultrawidify#175] [Q&A]
This wonderful naming has hit some popularity over [twitter](https://twitter.com/kaiyes_/status/1477981581381828609?s=28). I think everyone would like to know how did you come up with it.

https://github.com/tamius-han/ultrawidify/blob/4142dc1d0d1df8b0e5a991eeae1b9b9b4cd115c1/src/ext/lib/video-data/VideoData.ts#L72

**Comment by tamius-han** [comment]
Damn, I was wondering where all the attention came from, especially since the relevant subreddits (/r/ultrawidemasterrace, specifically) had nothing new today.

> I think everyone would like to know how did you come up with it.

Read through the thread, and the "someone raged at his machine" guesses are pretty on point. It's a reference to [Killing in the Name Of](https://www.youtube.com/watch?v=bWXazVhlyxQ). The reference is way too on-point to not include as an easter egg, even if I otherwise don't really consider myself a RatM fan.

### Bonus Content: Alternative Class Names

If I was doing the references today, I'd probably very briefly consider naming that class one of the following:

* `uw_ultrawidify_ate_my_pirate_css`
* `uw_ultrawidify_boss_mode`
* `uw_ultrawidify_tam_spaghetti` (not a rap fan, but this one would be a two-for-one deal in terms of memes)
* `uw_ultrawidify_i_am_the_resistance`
* `uw_ultrawidify_we_control`
* `uw_ultrawidify_css_we_believe_in`
* `uw_ultrawidify_shaping_of_the_video`
* `uw_ultrawidify_one_class_to_rule_them_all` (likely going all the way to `uw_ultrawidify_ash_bum_durbatuluk` as well, in a bid to Lurz all the fellow /r/lotrmemes nerds into a trap)

...  but then quickly decide that save for the last item, all of these are actually lame as fuck and far less apt than the 'Killing in the Name Of' reference and go with "fuck you and do what I tell you" anyway.

Though now that I'm at it, I might consider changing the name to one of these later, just to see if people will notice the reference the next time this surfaces.

### Bonus Content 2: Raging Against Which Machine, Exactly?

I see that people in that thread have a belief that I was raging at netflix specifically, but that's not the case. Getting things to work on Netflix was easy mode (except for autodetection, but that's not _wholly_ on netflix and is a topic for a different rant). I also noticed that netflix does try to correct aspect ratio on their own to some extent (as of one or two years ago I think).

So it wasn't really a rage against netflix.

It was more like a rage against [twitter and facebook](https://stuff.tamius.net/sacred-texts/2019/08/24/ultrawidify-the-twitchy-twitch-problem/), as I was fixing a problem created through the magic of ... less than stellar coding practices on my part. Frequenters of /r/badcode may find that blog post ... uh.
